pypsrp/client.py

```diff
@@ -109,17 +117,12 @@ def read_buffer(b_path, total_size, buffer_size):
             log.debug("Starting to send file data to remote process")
             powershell = PowerShell(pool)
             powershell.add_script(command).add_argument(dest)
-            powershell.invoke(input=read_gen)
-            log.debug("Finished sending file data to remote process")
+            powershell.invoke(input=read_gen, on_error_message="Failed to copy file")
```

Comment:
`on_error_message` is not a valid kwarg for [invoke](https://github.com/jborean93/pypsrp/blob/master/pypsrp/powershell.py#L1074) a PowerShell pipeline object.

I see you've added it but honestly I don't think it should be a common process. It's difficult because an error may be treated differently based on what the caller is expecting. PowerShell has terminating errors and non-terminating errors and the behaviour of the latter can be terminating depending on the `$ErrorActionPreference` setting. The [HadErrors](https://docs.microsoft.com/en-us/dotnet/api/system.management.automation.powershell.haderrors?view=pscore-6.2.0) is also a very weird property that I don't fully understand. It may be true depending on where the error/exception was raised but there are some cases where it has not but the caller may expect it to.

Personally I think the interface in `powershell.py` should try and match the actual .NET interface where possible to ease the transition for people who want to migrate their scripts from .NET to Python or vice versa. What you could do here is just create a common function in `client.py` to read the error's a PowerShell call and raise the exception. That way we have a common process abstracted out to a function but we keep the low level API more "pure".

Reply:
👍 I like the split. I will move it, thanks for the explanation